py/torch_tensorrt/dynamo/conversion/impl/elementwise/base.py

```diff
@@ -140,16 +141,45 @@ def convert_binary_elementwise(
 
     if trt_promoted_type != lhs_val.dtype:
         lhs_val = cast_trt_tensor(
-            ctx, lhs_val, trt_promoted_type, name, target, source_ir
+            ctx, lhs_val, trt_promoted_type, f"{name}_cast_lhs_val", target, source_ir
         )
     if trt_promoted_type != rhs_val.dtype:
         rhs_val = cast_trt_tensor(
-            ctx, rhs_val, trt_promoted_type, name, target, source_ir
+            ctx, rhs_val, trt_promoted_type, f"{name}_cast_rhs_val", target, source_ir
         )
 
-    lhs_val, rhs_val = broadcast(
-        ctx.net, lhs_val, rhs_val, f"{name}_lhs", f"{name}_rhs"
-    )
+    lhs_val_shape = lhs_val.shape
+    rhs_val_shape = rhs_val.shape
+    rank_diff = len(lhs_val_shape) - len(rhs_val_shape)
+    if rank_diff > 0:
+        rhs_val = impl.slice.expand(
+            ctx, target, source_ir, f"{name}_expand_rhs_val", rhs_val, lhs_val_shape
+        )
+    elif rank_diff < 0:
+        lhs_val = impl.slice.expand(
+            ctx, target, source_ir, f"{name}_expand_lhs_val", lhs_val, rhs_val_shape
+        )
+    else:
+        if tuple(lhs_val_shape) != tuple(rhs_val_shape):
+            sum_diff = sum(lhs_val_shape) - sum(rhs_val_shape)
+            if sum_diff > 0:
+                rhs_val = impl.slice.expand(
+                    ctx,
+                    target,
+                    source_ir,
+                    f"{name}_expand_rhs_val",
+                    rhs_val,
+                    lhs_val_shape,
+                )
+            elif sum_diff < 0:
+                lhs_val = impl.slice.expand(
+                    ctx,
+                    target,
+                    source_ir,
+                    f"{name}_expand_lhs_val",
+                    lhs_val,
+                    rhs_val_shape,
+                )
```

Comment:
Is it necessary to do more than append/prepend ones to the tensors? For instance, it seems that this code, on input Tensors of shape `[3, 1]` and `[1, 1]`, would make the latter `[3, 1]` as well. Is this newly required?